OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/VideoDownloadQualityDialogFragment.kt

```diff
@@ -0,0 +1,121 @@
+package org.edx.mobile.view.dialog
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.graphics.Typeface
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import androidx.appcompat.app.AlertDialog
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.DialogFragment
+import org.edx.mobile.R
+import org.edx.mobile.core.IEdxEnvironment
+import org.edx.mobile.databinding.VideoQualityDialogFragmentBinding
+import org.edx.mobile.model.video.VideoQuality
+import org.edx.mobile.module.analytics.Analytics
+import org.edx.mobile.module.prefs.LoginPrefs
+import org.edx.mobile.util.ResourceUtil
+import org.edx.mobile.view.adapters.VideoQualityAdapter
+
+
+class VideoDownloadQualityDialogFragment(
+    var environment: IEdxEnvironment,
+    var callback: IListDialogCallback
+) : DialogFragment() {
+
+    private var loginPref: LoginPrefs? = environment.loginPrefs
+    private val videoQualities: ArrayList<VideoQuality> = arrayListOf()
+    private lateinit var binding: VideoQualityDialogFragmentBinding
+
+    init {
+        videoQualities.addAll(VideoQuality.values())
+    }
+
+    companion object {
+        @JvmStatic
+        val TAG: String = VideoDownloadQualityDialogFragment::class.java.name
+
+        @JvmStatic
+        fun getInstance(
+            environment: IEdxEnvironment,
+            callback: IListDialogCallback
+        ): VideoDownloadQualityDialogFragment {
+            return VideoDownloadQualityDialogFragment(environment, callback)
+        }
+    }
+
+    private fun setupWindow() {
+//        setStyle(STYLE_NO_FRAME, android.R.style.Theme_Holo_Dialog)
+//        dialog?.window?.setBackgroundDrawableResource(android.R.color.transparent)
+    }
+
+    override fun onCreateView(
```

Comment:
this one too?

Reply:
if we removed this method, then the list item will not visible to the user. So we need this method.